### PR TITLE
dev-python/namespace-jaraco: new namespace package required by other jaraco-* packages

### PR DESCRIPTION
**Pull request by sbraz**
@mgorny can you please review that? It's my first time handling a namespace package.

```diff
--- jaraco-packaging-4.1.ebuild 2018-04-28 18:04:14.463510520 +0200
+++ jaraco-packaging-5.1.ebuild 2018-06-06 23:14:08.942856642 +0200
@@ -1,7 +1,7 @@
 # Copyright 1999-2018 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2

-EAPI=6
+EAPI=7

 PYTHON_COMPAT=( pypy{,3} python{2_7,3_{4,5,6}} )

@@ -14,20 +14,22 @@

 LICENSE="MIT"
 SLOT="0"
-KEYWORDS="alpha amd64 arm arm64 hppa ia64 ~mips ppc ppc64 sparc x86 ~amd64-fbsd"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~sparc ~x86 ~amd64-fbsd"
 IUSE="doc test"

-RDEPEND=">=dev-python/six-1.4[${PYTHON_USEDEP}]"
+RDEPEND="
+       >=dev-python/six-1.4[${PYTHON_USEDEP}]
+       dev-python/namespace-jaraco[${PYTHON_USEDEP}]
+"
 DEPEND="dev-python/setuptools[${PYTHON_USEDEP}]
        >=dev-python/setuptools_scm-1.9[${PYTHON_USEDEP}]
        doc? (
-               dev-python/rst-linker[${PYTHON_USEDEP}]
+               >=dev-python/rst-linker-1.9[${PYTHON_USEDEP}]
                dev-python/sphinx[${PYTHON_USEDEP}]
        )
        test? (
                ${RDEPEND}
                >=dev-python/pytest-2.8[${PYTHON_USEDEP}]
-               dev-python/pytest-runner[${PYTHON_USEDEP}]
        )
 "

@@ -46,3 +48,8 @@
        PYTHONPATH=. py.test -v -k "not test_revived_distribution" \
                || die "tests failed with ${EPYTHON}"
 }
+
+python_install_all() {
+       distutils-r1_python_install_all
+       find "${ED}" -name '*.pth' -delete || die
+}
```

```diff
--- jaraco-logging-1.5.ebuild	2017-07-23 23:47:32.203526991 +0200
+++ jaraco-logging-1.5.2.ebuild	2018-06-06 23:06:39.479582807 +0200
@@ -1,9 +1,9 @@
-# Copyright 1999-2017 Gentoo Foundation
+# Copyright 1999-2018 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=6
+EAPI=7
 
-PYTHON_COMPAT=( python{2_7,3_4,3_5} )
+PYTHON_COMPAT=( python{2_7,3_4,3_5} pypy{,3} )
 
 inherit distutils-r1
 
@@ -14,23 +14,45 @@
 
 LICENSE="MIT"
 SLOT="0"
-KEYWORDS="amd64 x86"
-IUSE="test"
+KEYWORDS="~amd64 ~x86"
+IUSE="doc test"
 
-RDEPEND="dev-python/six[${PYTHON_USEDEP}]
+RDEPEND="
+	dev-python/namespace-jaraco[${PYTHON_USEDEP}]
+	dev-python/six[${PYTHON_USEDEP}]
 	dev-python/tempora[${PYTHON_USEDEP}]
 "
-DEPEND="dev-python/setuptools[${PYTHON_USEDEP}]
+DEPEND="
+	dev-python/setuptools[${PYTHON_USEDEP}]
 	>=dev-python/setuptools_scm-1.9[${PYTHON_USEDEP}]
+	doc? (
+		>=dev-python/jaraco-packaging-3.2[${PYTHON_USEDEP}]
+		>=dev-python/rst-linker-1.9[${PYTHON_USEDEP}]
+		dev-python/sphinx[${PYTHON_USEDEP}]
+	)
 	test? (
 		${RDEPEND}
 		>=dev-python/pytest-2.8[${PYTHON_USEDEP}]
-		dev-python/pytest-runner[${PYTHON_USEDEP}]
 	)
 "
 
 S="${WORKDIR}/${MY_PN}-${PV}"
 
+python_compile_all() {
+	if use doc; then
+		cd docs || die
+		sphinx-build . _build/html || die
+		HTML_DOCS=( docs/_build/html/. )
+	fi
+}
+
 python_test() {
-	PYTHONPATH=. py.test || die "tests failed with ${EPYTHON}"
+	# Override pytest options to skip flake8
+	PYTHONPATH=. py.test --override-ini="addopts=--doctest-modules" \
+		|| die "tests failed with ${EPYTHON}"
+}
+
+python_install_all() {
+	distutils-r1_python_install_all
+	find "${ED}" -name '*.pth' -delete || die
 }
```

### CODE REVIEW

File: dev-python/jaraco-logging/jaraco-logging-1.5.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+HOMEPAGE="https://github.com/jaraco/jaraco.logging"
+SRC_URI="mirror://pypi/${PN:0:1}/${MY_PN}/${MY_PN}-${PV}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="doc test"
+
+RDEPEND="
+	dev-python/namespace-jaraco[${PYTHON_USEDEP}]
```
Comment (prometheanfire) [inline dev-python/jaraco-logging/jaraco-logging-1.5.2.ebuild]: Where in https://github.com/jaraco/jaraco.logging/blob/1.5.2/setup.py is this pulled in and how?

File: dev-python/jaraco-packaging/jaraco-packaging-5.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+SRC_URI="mirror://pypi/${PN:0:1}/${MY_PN}/${MY_PN}-${PV}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~mips ~ppc ~ppc64 ~sparc ~x86 ~amd64-fbsd"
+IUSE="doc test"
+
+RDEPEND="
+	>=dev-python/six-1.4[${PYTHON_USEDEP}]
+	dev-python/namespace-jaraco[${PYTHON_USEDEP}]
```
Comment (prometheanfire) [inline dev-python/jaraco-packaging/jaraco-packaging-5.1.ebuild]: same question here

File: dev-python/jaraco-logging/jaraco-logging-1.5.2.ebuild (hunk 1 above)
Comment (mgorny) [reply dev-python/jaraco-logging/jaraco-logging-1.5.2.ebuild]: Please Read The Friendly Manual: https://wiki.gentoo.org/wiki/Project:Python/Namespace_packages

**Comment by prometheanfire** [comment]
Ok, in that case I think it looks good, but I'm not familiar enough to actually leave my vote.

**Comment by mgorny** [comment]
FTR, we've figured out it's already been broken.

**Pull request closed without merging**